### PR TITLE
indirect logo

### PR DESCRIPTION
**Pull request by hcvcastro**
- wsd: add indirect request logo image
- browser: css: add '#lokit-logo' style to fetch logo.png
- wsd: cache logo image

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

File: browser/css/cool.css [hunk 1]
```diff
@@ -892,6 +892,18 @@ nav.spreadsheet-color-indicator ~ #sidebar-dock-wrapper {
 	background-size: 82px;
 	background-position: right center;
 }
+
```
Comment (timar) [inline browser/css/cool.css:895]: it's defined in branding.css

File: wsd/FileServer.cpp [hunk 2]
```diff
@@ -58,6 +59,11 @@ using Poco::Net::HTTPResponse;
 using Poco::Net::NameValueCollection;
 using Poco::Util::Application;
 
+#define DEFAULT_LOGO "https://www.libreoffice.org/themes/libreofficenew/img/logo.png"
```
Comment (timar) [inline wsd/FileServer.cpp:62]: wrong logo

**Review by timar — CHANGES_REQUESTED** [review]

**Review by mmeeks — CHANGES_REQUESTED** [review]
Some tweaks needed here.

**Comment by hcvcastro** [comment]
Changes done

**Review by mmeeks — APPROVED** [review]
Much better - if Ash is happy with the indirect thing which (I'd like to hope) is not necessary - then this looks good to me.

**Comment by hcvcastro** [comment]
Updated changes

File: wsd/COOLWSD.cpp [hunk 3]
```diff
@@ -2340,19 +2340,19 @@ void COOLWSD::innerInitialize(Application& self)
...
+    Server = Util::make_unique<COOLWSDServer>();
+
 #if !MOBILEAPP
     SavedClipboards = Util::make_unique<ClipboardCache>();
 
     LOG_TRC("Initialize FileServerRequestHandler");
     FileServerRequestHandler::initialize();
 #endif
 
-    WebServerPoll = Util::make_unique<TerminatingPoll>("websrv_poll");
```
Comment (mmeeks) [inline wsd/COOLWSD.cpp:2350]: Not so sure on this one - one of the things I like from a security perspective is having loaded all the files from the file-system, and having cached them internally before we start our various other polls on listening sockets and accept external messages.
Perhaps we should have a large comment // ---------------- from here on we start getting external messages ----------------
in there in the original place.
If we need to have another part of this doing later can we have it in a method after webserver init etc. which should happen after the fileserver setup.

File: wsd/FileServer.cpp [hunk 4]
```diff
@@ -527,7 +531,14 @@ void FileServerRequestHandler::handleRequest(const HTTPRequest& request,
             handleWopiRequest(request, requestDetails, message, socket);
             return;
         }
+#else
+        if (Util::startsWith(relPath, std::string("/browser/dist/remote/logo.svg")))
```
Comment (mmeeks) [inline wsd/FileServer.cpp:535]: I'd really prefer the same name lokit-extra-img.svg as we pass it through - renaming things as they go past makes things more complex to maintain I think.

**Review by mmeeks — COMMENTED** [review]
Nearly there ! =) a few comments, I guess Ash may have some more. Thanks for getting this over the line Henry !

**Comment by hcvcastro** [comment]
Updated changes

**Review by mmeeks — COMMENTED** [review]
Looks great to me - if Ash is happy with the socket/request stuff I'd love to get it in =)

**Comment by mmeeks** [comment]
Ash says: " yes, looks good. " =) thanks Henry.

**Comment by Ashod** [comment]
Thanks @hcvcastro, good stuff!

**Pull request merged**